plugin-build/src/main/kotlin/io/sentry/android/gradle/telemetry/SentryTelemetryService.kt

```diff
@@ -0,0 +1,474 @@
+                buildType,
+                tags,
+                cliVersion = BuildConfig.CliVersion
+            )
+        }
+
+        fun register(project: Project): Provider<SentryTelemetryService> {
+            return project.gradle.sharedServices.registerIfAbsent(
+                getBuildServiceName(SentryTelemetryService::class.java),
+                SentryTelemetryService::class.java
+            ) {}
+        }
+
+        private fun isExecAvailable(): Boolean {
+            return GradleVersions.CURRENT >= GradleVersions.VERSION_7_5
+        }
+
+        private fun extraTagsFromExtension(
+            project: Project,
+            extension: SentryPluginExtension
+        ): Map<String, String> {
+            val tags = mutableMapOf<String, String>()
+
+            tags.put("SENTRY_debug", extension.debug.get().toString())
+            tags.put(
+                "SENTRY_includeProguardMapping",
+                extension.includeProguardMapping.get().toString()
+            )
+            tags.put(
+                "SENTRY_autoUploadProguardMapping",
+                extension.autoUploadProguardMapping.get().toString()
+            )
+            tags.put("SENTRY_autoUpload", extension.autoUpload.get().toString())
+            tags.put("SENTRY_uploadNativeSymbols", extension.uploadNativeSymbols.get().toString())
+            tags.put(
+                "SENTRY_autoUploadNativeSymbols",
+                extension.autoUploadNativeSymbols.get().toString()
+            )
+            tags.put("SENTRY_includeNativeSources", extension.includeNativeSources.get().toString())
+            // TODO PII?
+            tags.put(
+                "SENTRY_ignoredVariants_set",
+                extension.ignoredVariants.get().isNotEmpty().toString()
+            )
+            // TODO PII?
+            tags.put(
+                "SENTRY_ignoredBuildTypes_set",
+                extension.ignoredBuildTypes.get().isNotEmpty().toString()
+            )
+            // TODO PII?
+            tags.put(
+                "SENTRY_ignoredFlavors",
+                extension.ignoredFlavors.get().isNotEmpty().toString()
+            )
+            tags.put(
+                "SENTRY_experimentalGuardsquareSupport",
+                extension.experimentalGuardsquareSupport.get().toString()
+            )
+            tags.put(
+                "SENTRY_tracing_enabled",
+                extension.tracingInstrumentation.enabled.get().toString()
+            )
+            tags.put(
+                "SENTRY_tracing_debug",
+                extension.tracingInstrumentation.debug.get().toString()
+            )
+            tags.put(
+                "SENTRY_tracing_forceInstrumentDependencies",
+                extension.tracingInstrumentation.forceInstrumentDependencies.get().toString()
+            )
+            tags.put(
+                "SENTRY_tracing_features",
+                extension.tracingInstrumentation.features.get().toString()
```

Comment:
idk, but this should be `joinToString` perhaps? But I'm not sure

Reply:
will check

Reply:
`"tracing_features": "[DATABASE, FILE_IO, OKHTTP, COMPOSE]"`

I think it's OK as is.